app/scripts/modules/core/serverGroup/details/multipleServerGroups.controller.js

```diff
@@ -29,14 +31,21 @@ module.exports = angular.module('spinnaker.core.serverGroup.details.multipleServ
         return descriptor;
       };
 
-      let confirm = (submitMethod, verbs) => {
+      let confirm = (submitMethodName, verbs) => {
         let descriptor = getDescriptor(),
             monitorInterval = this.serverGroups.length * 1000;
         let taskMonitors = this.serverGroups.map(serverGroup => {
+          let provider = serverGroup.provider || serverGroup.type;
+          let submitter = serviceDelegate.hasDelegate(provider, 'serverGroup.submitter') ?
```

Comment:
I'm wondering if it would be cleaner to have a per-provider `submitParamMixins`, which would be a map keyed by `submitMethodName`. Then we could just use the standard `serverGroupWriter` methods. This is too condensed, but it would look something like `serverGroupWriter[submitMethodName](serverGroup, app, angular.extend(params, providerMixins`